proposals/runtime_on_invalidated.md

```diff
@@ -0,0 +1,151 @@
+# Proposal: browser.runtime.onInvalidated event
+
+**Summary**
+
+`browser.runtime.onInvalidated`
+
+This event allows extension contexts which stay alive to be notified when the
+extension context get invalidated.
+
+**Document Metadata**
+
+**Author:** carlosjeurissen
+
+**Sponsoring Browser:** Chrome
+
+**Contributors:** bershanskiy
+
+**Created:** 2025-03-26
+
+**Related Issues:** https://github.com/w3c/webextensions/issues/138
+
+## Motivation
+
+### Objective
+
+This event allows extension contexts which stay alive to be notified when the
```

Comment:
"extension context" is very generic, while this seems to primarily be targeting content scripts. Can we restrict this to content scripts?

Reply:
@Rob--W it is generic on purpose as to cover both content scripts and iframed extension contexts on third party websites. These iframes stay alive on some browsers and require this event to handle their state. The objective states `contexts which stay alive` to exclude extension contexts which get killed on invalidation.